x-pack/plugins/synthetics/server/synthetics_service/private_location/synthetics_private_location.ts

```diff
@@ -73,14 +80,19 @@ export class SyntheticsPrivateLocation {
         const location = privateLocations?.find((loc) => loc.id === privateLocation.id)!;
         const newPolicy = await this.generateNewPolicy(config, location);
 
+        if (!newPolicy) {
+          return null;
```

Comment:
In general, from synthetics_private_location, I think we need to bubble errors back up to the caller to ensure that error tooltips appear appropriately when there are errors saving config. Perhaps that can be done in a follow up.

I added this here and some try catches around fleet functions because some errors were unhandled and causing Kibana to crash entirely.